extensions-core/druid-bloom-filter/src/main/java/org/apache/druid/query/aggregation/bloom/BloomFilterAggregatorFactory.java

```diff
@@ -0,0 +1,233 @@
+  {
+    ColumnSelectorPlus<BloomFilterAggregatorColumnSelectorStrategy> selectorPlus =
+        DimensionHandlerUtils.createColumnSelectorPlus(
+            STRATEGY_FACTORY,
+            field,
+            columnFactory
+        );
+
+    return new BloomFilterAggregator(selectorPlus, maxNumEntries);
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(ColumnSelectorFactory columnFactory)
+  {
+    ColumnSelectorPlus<BloomFilterAggregatorColumnSelectorStrategy> selectorPlus =
+        DimensionHandlerUtils.createColumnSelectorPlus(
+            STRATEGY_FACTORY,
+            field,
+            columnFactory
+        );
+
+    return new BloomFilterBufferAggregator(selectorPlus, maxNumEntries);
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    // idk how to compare?
+    return (Comparator<BloomKFilter>) (o1, o2) -> 0;
+  }
+
+  @Override
+  public Object combine(Object lhs, Object rhs)
+  {
+    if (rhs == null) {
+      return lhs;
+    }
+    if (lhs == null) {
+      return rhs;
+    }
+    ((BloomKFilter) lhs).merge((BloomKFilter) rhs);
+    return lhs;
+  }
+
+  @Override
+  public AggregatorFactory getCombiningFactory()
+  {
+    return new BloomFilterMergeAggregatorFactory(name, name, maxNumEntries);
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Collections.singletonList(new BloomFilterAggregatorFactory(name, field, maxNumEntries));
+  }
+
+  @Override
+  public Object deserialize(Object object)
+  {
+    final ByteBuffer buffer;
+
+    if (object instanceof byte[]) {
+      buffer = ByteBuffer.wrap((byte[]) object);
+    } else if (object instanceof ByteBuffer) {
+      // Be conservative, don't assume we own this buffer.
+      buffer = ((ByteBuffer) object).duplicate();
+    } else if (object instanceof String) {
+      buffer = ByteBuffer.wrap(Base64.decodeBase64(StringUtils.toUtf8((String) object)));
+    } else {
+      return object;
+    }
+
+    ByteBufferInputStream byteBufferInputStream = new ByteBufferInputStream(buffer);
+    try {
+      return BloomKFilter.deserialize(byteBufferInputStream);
+    }
+    catch (Exception ex) {
+      throw new RuntimeException("Failed to deserialize bloomK filter", ex);
+    }
+  }
+
+  @Override
+  public Object finalizeComputation(Object object)
+  {
+    return object;
+  }
+
+  @JsonProperty
+  @Override
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty
+  public DimensionSpec getField()
+  {
+    return field;
+  }
+
+  @JsonProperty
+  public int getMaxNumEntries()
+  {
+    return maxNumEntries;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    return Collections.singletonList(field.getDimension());
+  }
+
+  @Override
+  public String getTypeName()
+  {
+    return "bloomFilter";
+  }
+
+  @Override
+  public int getMaxIntermediateSize()
+  {
+    BloomKFilter throwaway = new BloomKFilter(maxNumEntries);
```

Comment:
Cache the value in a static, don't create a throwaway each time.

Reply:
What sort of cache is most appropriate? A static int2object map? caffeine?

Reply:
"cache" = set to a static final field, I meant here

Reply:
I don't think a static works since the result depends on `maxNumEntries` which is a query time parameter. If this method gets called multiple times i can set an instance field to this value at constructor time, or I can see if the math that computes the size of the long array inside is accessible to call that directly.

Reply:
Direct computation, if not supported by the library directly (maybe we could contribute that?) could be dangerous if the algorithm in the library changes in some version.

I think most of the time only one `maxNumEntries` value will be seen per JVM, small Int2Object map (stopped being populated after say 10 entires) should work

Reply:
*Int2IntMap, you could cache the final values.

Reply:
Added a method to our copied `BloomKFilter` to compute the size required given a number of entries, avoiding this throwaway